pkg/util/util_test.go

```diff
@@ -1571,49 +1573,69 @@ func TestIsValidProjectDir(t *testing.T) {
 	}
 }
 
-/*
 func TestGetGitHubZipURL(t *testing.T) {
+	commitid := "39b5ec7833a65278012eab13a895471bba2cd03d"
+	tag := "1.0.0"
 	tests := []struct {
-		name          string
-		zipURL        string
-		expectedError string
+		name           string
+		project        common.DevfileProject
+		expectedErrors []string
 	}{
 		{
-			name:          "Case 1: Invalid http request",
-			zipURL:        "http://github.com/che-samples/web-nodejs-sample/archive/master",
-			expectedError: "Invalid GitHub URL. Please use https://",
-		},
-		{
-			name:          "Case 2: Invalid owner",
-			zipURL:        "https://github.com//web-nodejs-sample/archive/master",
-			expectedError: "Invalid GitHub URL: owner cannot be empty. Expecting 'https://github.com/<owner>/<repo>'",
+			name: "Case 1: Invalid http request",
+			project: common.DevfileProject{
+				Source: common.DevfileProjectSource{
+					Location: "http://github.com/che-samples/web-nodejs-sample/archive/master",
+				},
+			},
+			expectedErrors: []string{"Invalid GitHub URL. Please use https://"},
 		},
 		{
-			name:          "Case 3: Invalid repo",
-			zipURL:        "https://github.com/che-samples//archive/master",
-			expectedError: "Invalid GitHub URL: repo cannot be empty. Expecting 'https://github.com/<owner>/<repo>'",
+			name: "Case 2: Invalid owner",
+			project: common.DevfileProject{
+				Source: common.DevfileProjectSource{
+					Location: "https://github.com//web-nodejs-sample/archive/master",
+				},
+			},
+			expectedErrors: []string{"Invalid GitHub URL: owner cannot be empty. Expecting 'https://github.com/<owner>/<repo>'"},
 		},
 		{
-			name:          "Case 4: Non-existent URL",
-			zipURL:        "https://github.com/this/does/not/exist",
-			expectedError: "Error getting zip url. Response: 404 Not Found.",
+			name: "Case 3: Invalid repo",
+			project: common.DevfileProject{
+				Source: common.DevfileProjectSource{
+					Location: "https://github.com/che-samples//archive/master",
+				},
+			},
+			expectedErrors: []string{"Invalid GitHub URL: repo cannot be empty. Expecting 'https://github.com/<owner>/<repo>'"},
 		},
 		{
-			name:          "Case 5: Valid SSH Github URL",
-			zipURL:        "git@github.com:che-samples/web-nodejs-sample.git",
-			expectedError: "",
+			name: "Case 4: Invalid HTTPS Github URL with tag and commit",
```

Comment:
I can see only the negative test cases has been covered. Can you please add some more unit test case by considering positive scenarios also?

Reply:
I had those originally but since we don't want any external dependencies on test cases I cannot test the positive scenarios. The function calls the GithubAPI to retrieve the zip url.

Reply:
I am not sure if we can mock those `GithubAPI`, Anway I am seeking suggestion from @mik-dass  @girishramnani  @kadel  on how to make it happen.  @EnriqueL8  please feel free to add your suggestion too.

Reply:
I think we can mock the APIs using httptest package. Here's how they did the mocking upstream https://github.com/google/go-github/blob/master/github/repos_contents_test.go#L372 and here's how they setup up the test server https://github.com/google/go-github/blob/master/scrape/scrape_test.go#L14.

Reply:
I think that is a good idea. I would need to change the function definition to accept the client instance used to query the server. I think a bigger question comes to mind, which is by adding these test we will be testing the go-github package and my function just calls that api and returns. So what benefit would we gain from testing here, over it being already tested in the package?

Reply:
Agreed. I think we need to find a way to mock the library. Maybe this discussion might help https://github.com/google/go-github/issues/113

Reply:
@mik-dass Thanks for the link to that thread! I tried to implement the mocking solution they propose in https://github.com/gaia-pipeline/gaia/blob/69b69b1186c484a55133f58ff0919151baa79dfd/workers/pipeline/git.go#L211 by doing: 

```
type GithubRepoService interface {
	GetArchiveLink(ctx context.Context, owner, repo string, archiveformat archiveFormat, opts *github.RepositoryContentGetOptions, followRedirects bool)(*url.URL, *github.Response, error)
}
```
Unfortunately the type `archiveFormat` in the function definition is internal, so this interface cannot be used for both the Github Client Repo struct and the mocked version similar to the previous implementation.  I am not sure there is way to solve it without asking the library to make that definition  public. I tried putting a `string` type, because the actual type is defined as `type archiveFormat string` but that didn't work.

Reply:
OK let's leave it for now.